gcp-autocomplete-lib.js

```diff
@@ -0,0 +1,107 @@
+const GCCompute = require("@google-cloud/compute");
+const { ProjectsClient } = require("@google-cloud/resource-manager");
+const _ = require("lodash");
+
+const {
+  RESOURCE_OPERATIONS,
+} = require("./gcp-lib");
+const parsers = require("./parsers");
+const helpers = require("./helpers");
+
+function createListItemsFunction(clientClass, fields) {
+  return async function listGcpItems(query, pluginSettings, pluginParams) {
+    const { result } = await getListResults(fields, pluginSettings, pluginParams, clientClass);
+    return result;
+  };
+}
+
+async function listGcpProjects(query, pluginSettings, pluginParams) {
+  const { result } = await getProjects(["projectId", "displayName"], pluginSettings, pluginParams);
+  return result;
+}
+
+async function listGcpRegions(query, pluginSettings, pluginParams) {
+  const { result } = await getListResults(["name"], pluginSettings, pluginParams, GCCompute.RegionsClient);
+  return result;
+}
+
+async function listGcpZones(query, pluginSettings, pluginParams) {
+  const { result, params } = await getListResults(["name"], pluginSettings, pluginParams, GCCompute.ZonesClient);
+
+  const region = parsers.autocomplete(params.region);
```

Comment:
`getListResults` calls `helpers.mapAutoParams` which parsers every param with `parsers.autocomplete`, no need to repeat that here.

Reply:
Redesigned